psi4/driver/driver_findif.py

```diff
@@ -1017,65 +1126,371 @@ def init_string(data):
+        reference = self.findifrec["reference"]
+        task = results_list["reference"]
+        response = task.return_result
+        reference["module"] = getattr(task.provenance, "module", None)
+
+        if task.driver == 'energy':
+            reference['energy'] = response
+
+        elif task.driver == 'gradient':
+            reference['gradient'] = response
+            reference['energy'] = task.extras['qcvars']['CURRENT ENERGY']
+
+        elif task.driver == 'hessian':
+            reference['hessian'] = response
+            reference['energy'] = task.extras['qcvars']['CURRENT ENERGY']
+            if 'CURRENT GRADIENT' in task.extras['qcvars']:
+                reference['gradient'] = task.extras['qcvars']['CURRENT GRADIENT']
+
+        dipole_available = False
+        if 'CURRENT DIPOLE' in task.extras['qcvars']:
+            reference['dipole'] = task.extras['qcvars']['CURRENT DIPOLE']
+            dipole_available = True
+
+        # load AtomicComputer results into findifrec[displacements]
+        for label, displacement in self.findifrec["displacements"].items():
+            task = results_list[label]
+            response = task.return_result
+
+            if task.driver == 'energy':
+                displacement['energy'] = response
+
+            elif task.driver == 'gradient':
+                displacement['gradient'] = response
+                displacement['energy'] = task.extras['qcvars']['CURRENT ENERGY']
+
+            elif task.driver == 'hessian':
+                displacement['hessian'] = response
+                displacement['energy'] = task.extras['qcvars']['CURRENT ENERGY']
+                if 'CURRENT GRADIENT' in task.extras['qcvars']:
+                    displacement['gradient'] = task.extras['qcvars']['CURRENT GRADIENT']
+
+            if 'CURRENT DIPOLE' in task.extras['qcvars']:
+                displacement['dipole'] = task.extras['qcvars']['CURRENT DIPOLE']
+
+        # apply finite difference formulas and load derivatives into findifrec[reference]
+        if self.metameta['mode'] == '1_0':
+            G0 = assemble_gradient_from_energies(self.findifrec)
+            self.findifrec["reference"][self.driver.name] = G0
+
+        elif self.metameta['mode'] == '2_1':
+            DD0 = assemble_dipder_from_dipoles(self.findifrec, self.metameta['irrep'])
+            self.findifrec["reference"]["dipole derivative"] = DD0
+
+            H0 = assemble_hessian_from_gradients(self.findifrec, self.metameta['irrep'])
+            self.findifrec["reference"][self.driver.name] = H0
+
+        elif self.metameta['mode'] == '2_0':
+            try:
+                G0 = assemble_gradient_from_energies(self.findifrec)
+            except KeyError:
+                core.print_out("Unable to construct reference gradient from Hessian displacements.")
+                # TODO: this happens properly when a subset irrep doesn't
+                #  have the right displacements for grad. For both this case
```

Comment:
"subset irrep doesn't have the right displacements for grad" is opaque. I think you mean "when requested symmetry block of displacements don't have the totally symmetric gradient displacements"?